Pass context to conditional function in "maybe" clause

sboehler opened the pull request:
Use case: Implementing a "get" method which fetches an object by certain parameters. 

- An admin should be able to fetch any object by id, irrespective of the owner. So { id } shall be a valid params object
- However, a normal user shall only be allowed to issue the request scoped to its own objects. So { id, user_id } shall be a valid params object and { id } should be invalid.

I am passing the user in the context object to the rules.run method. However, currently the context can't be evaluated in the conditional function of the 'maybe' clause.

This small pull request enhances the signature of the conditional function and allows the above use case to be implemented as:

```
const rules = new Checkit({
    id: ['integer'],
    user_id: ['integer', (val, context) => context.user.admin || context.user.id === val],
    name: ['string']
})
.maybe({ user_id: 'required' }, (id, context) => !context.user.admin);
```

Then, validation is run using 

```
paramRules.run(params, { user })
.then(...)
```

Let me know what you think, I appreciate any feedback!

rhys-vdw commented:
Yep. This makes sense. Could you add a test and include some documentation in the README please?

sboehler commented:
Thanks @rhys-vdw for the quick reply. I have updated my branch with a test (both run and runSync) and updated the README. Let me know if you need anything more.

rhys-vdw commented:
Thanks @sboehler. Looks good!
